bundles/org.openhab.binding.wundergroundupdatereceiver/src/main/resources/OH-INF/thing/channel-group-types.xml

```diff
@@ -0,0 +1,118 @@
+<?xml version="1.0" ?>
+<thing:thing-descriptions bindingId="wundergroundupdatereceiver"
+	xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+	xmlns:thing="https://openhab.org/schemas/thing-description/v1.0.0"
+	xsi:schemaLocation="https://openhab.org/schemas/thing-description/v1.0.0 https://openhab.org/schemas/thing-description-1.0.0.xsd">
+
+	<channel-group-type id="metadata-group">
```

Comment:
the list of channels is quite long. can you mark minor important ones as `advanced`? e.g. `Pollution `group

My WH2600 station with indoor and outdoor sensor is reporting the following additional values:
- windchillf
- weeklyrainin
- monthlyrainin
- yearlyrainin
- lowbatt 0/1 -> OFF/ON
- action
- realtime 0/1 -> OFF/ON
- rtfreq
- ID
Can they be added as channel as well?

full example:
`ID=KAZPHOEN11&PASSWORD=xxx&tempf=35.2&humidity=97&dewptf=34.5&windchillf=35.2&winddir=294&windspeedmph=2.46&windgustmph=2.46&rainin=0.00&dailyrainin=0.00&weeklyrainin=0.17&monthlyrainin=1.82&yearlyrainin=4.70&solarradiation=43.31&UV=0&indoortempf=72.3&indoorhumidity=36&baromin=29.84&dateutc=2021-02-20%207:27:47&softwaretype=WH2600%20V2.2.8&action=updateraw&realtime=1&rtfreq=5 `

Reply:
action, realtime and rtfreq are only of interest for the wunderground.com protocol, so I chose to exclude them. action is also always 'updateraw' as far as I can tell from the docs. ID is the representation-property stationId.
I don't seem to have a lowbat parameter, althoug it is in the user interface and would be useful in the thing. Does it only show up in particular requests? It is irrelevant for wunderground.com though.
There is another unofficial binding for this weather station software that scrapes the web ui and includes the battery state if I am not mistaken.

Reply:
Yes, ID is the station ID and it would be nice to have it in the data of a channel as well, i.e. if you own several stations.

There seem to be stations, that report `lowbatt` via the GET request:
`weatherstation/updateweatherstation.php?ID=YYYYYYYY&PASSWORD=XXXXXX&tempf=70.7&humidity=41&dewptf=45.9&windchillf=70.7&winddir=268&windspeedmph=0.00&windgustmph=0.00&rainin=0.00&dailyrainin=0.00&weeklyrainin=0.00&monthlyrainin=0.00&yearlyrainin=0.00&solarradiation=0.00&UV=0&indoortempf=69.8&indoorhumidity=40&baromin=28.94&lowbatt=0&dateutc=2018-12-25%206:40:24&softwaretype=WH2600GEN_V2.2.5&action=updateraw&realtime=1&rtfreq=5`

My last station and it's firmware did. My latest does not.
We should take it from the GET request, if available. I do not see the immediate need to parse the website.

`rtfreq` is interesting in order to know the expected frequency of the update. I reset my station, if I see that the actual cycle gets out of range.

Reply:
Ok. I'll take your word for it and add lowbatt as it's very useful when present.
I wouldn't trust rtfreq on the other hand. In my logs the weather station made requests every 15s even though rtfreq is set to 5. Do you mean the station slows down, so you would make a rule that sends a restart station request if rtfreq and the actual frequency differ by too much?
The ID is what you enter in the thing configuration, so you have one thing per wunderground ID. You can then have multiple instruments submitting measurements to the same wunderground ID. In a rule you can look up the stattionId from the thing or channel. It seems redundant to include a channel with a constant value that equals the representation property of the thing.

Reply:
> Ok. I'll take your word for it and add lowbatt as it's very useful when present.
👍 

> I wouldn't trust rtfreq on the other hand. In my logs the weather station made requests every 15s even though rtfreq is set to 5. Do you mean the station slows down, so you would make a rule that sends a restart station request if rtfreq and the actual frequency differ by too much?
Sure, `rtfreq` is also not reliable for my device.
It is sending every 16s. After some month of operation, the cycletime increases. If the average is above a certain limit, I send a GET request in order to reset the device before it hangs completely.
`sendHttpGetRequest("http://192.168.69.55/msgreboot.htm")`

`The ID is what you enter in the thing configuration, so you have one thing per wunderground ID. You can then have multiple instruments submitting measurements to the same wunderground ID. In a rule you can look up the stattionId from the thing or channel. It seems redundant to include a channel with a constant value that equals the representation property of the thing.`
Ok. I never accessed things or channels from rules. Only items so far. I will figure that out, if there is really a need.

Reply:
You can manually add a Number item, called let's say "heatindex". Then you can update it in a rule that triggers on let's say the humidity item being updated and put in a dsl rule like
```
val RH = WundergroundUpdateReceiverBindingThing_Humidity.getStateAs(DecimalType).floatValue()
val T = WundergroundUpdateReceiverBindingThing_OutdoorTemperature.getStateAs(DecimalType).floatValue()
val hix = -42.379 + 2.04901523*T + 10.14333127*RH - 0.22475541*T*RH - 0.00683783*T*T - 0.05481717*RH*RH + 0.00122874*T*T*RH + 0.00085282*T*RH*RH - 0.00000199*T*T*RH*RH
heatindex.setState(new DecimalType(hix))
```

You would probabæy also have some if/then stuff for the adjustments but this is just an example.
Similar rules can also be used to calculate the averages for example thusly:
`logInfo('HeatIndexRule', 'Average last 2 min [{}]', WundergroundUpdateReceiverBindingThing_CurrentWindSpeed.averageSince(ZonedDateTime.now().plusMinutes(-2)))`

Reply:
Is there a pre-defined order how the different channels are updated? I guess not, as also the openhab framework in between will do it's magic.
When using humidity as the trigger, the chance is 50:50 that both values used for the calculation origin from the same data sample.

Reply:
Example for README:
```
rule "Calculate HeatIndex"
when
    Item WH2k6_OutdoorTemperature changed or
    Item WH2k6_Humidity changed
then
    if ( (WH2k6_OutdoorTemperature.state != NULL) && (WH2k6_OutdoorTemperature.state != UNDEF) &&
         (WH2k6_Humidity.state != NULL)           && (WH2k6_Humidity.state != UNDEF)
       )
    {
        val Double tempf    = (WH2k6_OutdoorTemperature.state as QuantityType<Number>).toUnit("°F").doubleValue
        val Double humidity = (WH2k6_Humidity.state           as QuantityType<Number>).toUnit("%").doubleValue

        // https://www.wpc.ncep.noaa.gov/html/heatindex_equation.shtml
        if ( tempf>=80.0 ) {
            var Double heatindex = -42.379 + 2.04901523*tempf + 10.14333127*humidity - 0.22475541*tempf*humidity - 0.00683783*tempf*tempf - 0.05481717*humidity*humidity + 0.00122874*tempf*tempf*humidity + 0.00085282*tempf*humidity*humidity - 0.00000199*tempf*tempf*humidity*humidity;

            if ( (humidity <= 13.0) && (tempf >= 80.0) && (tempf <= 112.0) ) {
                heatindex -= ((13.0 - humidity)/4.0) * (Math::sqrt(17.0-Math::abs(tempf-95.0)/17.0));
            } else if ( (humidity >= 85.0) && (tempf >= 80.0) && (tempf <= 87.0) )  {
                heatindex += ((humidity - 85.0)/10.0)*((87.0-tempf)/5.0);
            }
            WH2k6_HeatIndex.postUpdate(heatindex + "°F")
        }
        else
        {
            WH2k6_HeatIndex.postUpdate(UNDEF)
        }
    }
end
```

Reply:
The channels are updated in the order they appear in the GET request from the weather-station. But yes, it might be easiest to trigger the calculations from the trigger channel as the "forward to WU" example does. But all this is standard openhab rule programming.
But OK, I have also found it difficult to find good examples of the rule DSL syntax...